### PR TITLE
(DOCSP-27941) Flutter: add Decimal128 support to docs

### PR DESCRIPTION
**Pull request by krollins-mdb**
### Jira

- https://jira.mongodb.org/browse/DOCSP-27941

### Staged Changes

- [Data Types](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27941/sdk/flutter/realm-database/model-data/data-types/#decimal128)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Comment by krollins-mdb** [comment]
@desistefanova, could you make sure that I captured this information correctly?

**Review by desistefanova — APPROVED** [review]
LGTM.

File: source/sdk/flutter/realm-database/model-data/data-types.txt [hunk 1]
```diff
@@ -275,6 +275,28 @@ such as ``Uuid.v4()``.
...
+
+Dart doesn't have a native decimal type. You can use 
+`Decimal128 <https://pub.dev/documentation/realm/latest/realm/Decimal128-class.html>`__, 
+which is a 128-bit MongoDB-specific implementation of :wikipedia:`IEEE-754 <IEEE_754>`.
+When defining a decimal type, use the ``Decimal128`` BSON type.
+
+Because Dart doesn't have a native decimal type, you might encounter odd edge
+case behavior. For example, the Dart ``compareTo()`` method implements total
+ordering that mimics the Dart ``double`` type. This means the following things
+are true when using ``compareTo()``:
```
Comment (nielsenko) [inline source/sdk/flutter/realm-database/model-data/data-types.txt:291]: This is not odd. This is how Decimal128 should behave as defined by IEEE-754

File: source/sdk/flutter/realm-database/model-data/data-types.txt [hunk 2]
```diff
@@ -275,6 +275,28 @@ such as ``Uuid.v4()``.
...
    :language: dart
 
+.. _flutter-decimal128:
+
+Decimal128
+~~~~~~~~~~
+
+Dart doesn't have a native decimal type. You can use 
+`Decimal128 <https://pub.dev/documentation/realm/latest/realm/Decimal128-class.html>`__, 
+which is a 128-bit MongoDB-specific implementation of :wikipedia:`IEEE-754 <IEEE_754>`.
```
Comment (nielsenko) [inline source/sdk/flutter/realm-database/model-data/data-types.txt:285]: Not really anything MongoDB specific about it. We are basically just wrapping the intel implementation that is used in both realm-core and mongodb.

File: source/sdk/flutter/realm-database/model-data/data-types.txt (hunk 1 above)
Comment (nielsenko) [reply source/sdk/flutter/realm-database/model-data/data-types.txt:291]: What is a bit odd is that realm-core actually has a different total ordering .. now thats weird, but I don't think we should mention this in the dart docs.

**Review by nielsenko — CHANGES_REQUESTED** [review]

File: source/sdk/flutter/realm-database/model-data/data-types.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/sdk/flutter/realm-database/model-data/data-types.txt:291]: Thank you! Yeah, I shouldn't have characterized that as odd. I'll elaborate on the point a bit.

File: source/sdk/flutter/realm-database/model-data/data-types.txt (hunk 2 above)
Comment (krollins-mdb) [reply source/sdk/flutter/realm-database/model-data/data-types.txt:285]: Today I learned. Thanks for the info! I'll update to remove "MongoDB-specific".

**Review by dacharyc — APPROVED** [review]
LGTM

**Pull request merged**